pkg/clustermanager/provision_node.go

```diff
@@ -139,7 +152,7 @@ func (provisioner *NodeProvisioner) prepareDocker() error {
 	// docker-ce
 	aptPreferencesDocker := `
 Package: docker-ce
-Pin: version 17.03.*
+Pin: version 18.06.0~ce~3-0~ubuntu
```

Comment:
I had issues with this one. It still installed 18.06.1 in my case. Did that change to you? 

Anyway, I'm not sure if this approach is actually a good practice. It's the first and only thing which actually worked

Reply:
I believe it worked for me. There was no problems with the containers, I also tested with rook.io storage and everything worked fine. 

Which problems did you face with this one?

Reply:
It failed during kubeadm test phase to talk to the container runtime, as it didn't work to communicate. It is a known issue for 18.06.1. That actually makes me super-skeptic, as I'm not convinced if that approach is the right one. It's still the only working solution to make k8s run on Ubuntu 18 with a supported runtime.

Also interesting if 18.09 is already available and if this approach is working with it.

Could you please check that?

Reply:
18.09 has a problem with crictl binary not understand the new API. But 18.06 works fine.

Reply:
Just to review, with this PR you definitely get 18.06.0 and not 18.06.1, which is not working?

If yes, I'm OK with it

Reply:
I am not sure about 18.06.1, but 18.06.0 is working and 18.09 is not, according to my tests.